test/common/object_pool_test.cpp

```diff
@@ -34,6 +34,46 @@ TEST(ObjectPoolTests, SimpleReuseTest) {
   }
 }
 
+// Allocate more memory space than the object pool and expect exceptions
+// NOLINTNEXTLINE
+TEST(ObjectPoolTests, ExceedLimitTest) {
+  const uint32_t repeat = 1;
+  const uint64_t size_limit = 10;
+  const uint64_t reuse_limit = 10;
```

Comment:
```suggestion
  const uint64_t reuse_limit = size_limit;
```

Nit: maybe makes it slightly clearer that the value of this variable isn't relevant to what is being tested?